packages/cli/src/lib/deps-optimizer.ts

```diff
@@ -0,0 +1,181 @@
+import {AbortError, BugError} from '@shopify/cli-kit/node/error';
+import {extname} from '@shopify/cli-kit/node/path';
+import {renderFatalError} from '@shopify/cli-kit/node/ui';
+import colors from '@shopify/cli-kit/node/colors';
+import {type FormatOptions} from './format-code.js';
+import {importLangAstGrep} from './ast.js';
+import {replaceFileContent} from './file.js';
+import {outputInfo} from '@shopify/cli-kit/node/output';
+
+const throttledOptimizableDeps = new Set<string>();
+let debouncedBannerTimeout: NodeJS.Timeout | undefined;
+
+export function createEntryPointErrorHandler({
+  disableDepsOptimizer,
+  showSuccessBanner,
+  configFile,
+  formatOptionsPromise,
+}: {
+  showSuccessBanner: () => void;
+  disableDepsOptimizer?: boolean;
+  configFile?: string;
+  formatOptionsPromise: Promise<FormatOptions>;
+}) {
+  return async function entryPointErrorHandler({
+    optimizableDependency,
+    stack,
+  }: {
+    optimizableDependency?: string;
+    stack: string;
+  }) {
+    const message = stack.split('\n')[0] ?? stack;
+    const cleanStack = stack
+      .split('\n')
+      .filter((line) => !line.includes('virtual:remix'))
+      .join('\n');
+
+    const headline = 'MiniOxygen errored while running your entrypoint';
+
+    if (optimizableDependency) {
+      if (disableDepsOptimizer || !configFile) {
+        const depError = new BugError(
+          `${headline}:\n\n${colors.dim(message)}`,
+          `Try adding '${colors.yellow(
+            optimizableDependency,
+          )}' to your Vite config\'s ssr.optimizeDeps.include`,
+        );
+        depError.stack = cleanStack;
+        renderFatalError(depError);
+      } else if (!throttledOptimizableDeps.has(optimizableDependency)) {
+        // When multiple requests hit the same error, we only want to
+        // to log and add the dependency to the Vite config once.
+        throttledOptimizableDeps.add(optimizableDependency);
+        setTimeout(
+          () => throttledOptimizableDeps.delete(optimizableDependency),
+          2000,
+        );
+
+        addToViteOptimizeDeps(
+          optimizableDependency,
+          configFile,
+          await formatOptionsPromise,
+          cleanStack,
+        )
+          .then(() => {
+            setTimeout(() => {
+              outputInfo(
+                `\nAdded '${colors.yellow(
+                  optimizableDependency,
+                )}' to your Vite config's ssr.optimizeDeps.include\n`,
```

Comment:
This log line is shown **after** we add it, though. If we try to log it before or during, it will be hidden by Vite's "clear screen after restart logic".
Also, it's always printed after Vite's `[vite] server restarted` log. That's why I wrote `Added` in the first place.

With this info, can you confirm if we still want `Adding`?